nullaway/src/main/java/com/uber/nullaway/GenericsChecks.java

```diff
@@ -787,6 +787,18 @@ public static Nullness getGenericReturnNullnessAtInvocation(
     if (!(tree.getMethodSelect() instanceof MemberSelectTree) || invokedMethodSymbol.isStatic()) {
       return Nullness.NONNULL;
     }
+    Tree expressionTree = ((MemberSelectTree) tree.getMethodSelect()).getExpression();
+    if (expressionTree instanceof NewClassTree
+        && ((NewClassTree) expressionTree).getIdentifier() instanceof ParameterizedTypeTree) {
+      ParameterizedTypeTree paramTypedTree =
+          (ParameterizedTypeTree) ((NewClassTree) expressionTree).getIdentifier();
+      // The case of having a diamond operator
+      if (paramTypedTree.getTypeArguments().isEmpty()) {
+        // bail out
+        // TODO: support diamond operators
+        return Nullness.NONNULL;
+      }
+    }
     Type methodReceiverType =
         castToNonNull(
             getTreeType(((MemberSelectTree) tree.getMethodSelect()).getExpression(), state));
```

Comment:
Instead of adding all the code above, why not just store the result of this `getTreeType()` call in a local variable, and if it is `null`, return `Nullness.NONNULL`?  That is more general I think.

Reply:
I agree, I added it in the same condition, does this work?

Reply:
Better to do it after that condition and store the result of `getTreeType()` in a local variable, so we don't call it twice.  Otherwise the logic looks good.

Reply:
I've removed the redundant call